src/array.jl

```diff
@@ -212,6 +212,31 @@ end
 
             convert($A{T, N, R}, A)
         end
+
+        function Base.vcat{T,N,R}(A1::$A{T, N, R}, An::$A...)
+            As = (A1, An...)
+            levels = unique(T[[a.pool.levels for a in As]...;])
```

Comment:
Is this sufficient? https://github.com/nalimilan/CategoricalArrays.jl/pull/18/files#diff-70b907e278a1699afcee58e1bf1c41ecR555